### PR TITLE
Add correct aria attributes to dropdown

### PR DESCRIPTION
**Pull request by RVMendoza**
## What does this PR do?
Fix accessibility problems with dropdown for all Implementations.

The main problem is that we were using the correct aria labels for popup, hence the many violations.
Dropdown has been refactored according to [W3C Examples](https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html)

To fully fix the accessibility violations, we had to add aria-labels and give a reasonable default when Props/Inputs like `screenReaderText`,`title`, or `triggerText` were not provided.

The image below is the accessibility violations according to Storybook before the changes:
image

### Associated Issue 
#2390 

### Documentation
No updates

### Code
 - [x] Build Component in Spark Vanilla (Sass, HTML, JS)
 - [x] Build Component in Spark Angular
 - [x] Build Component in Spark React
 - [x] Unit Testing in Spark Vanilla with `npm run test` (100% coverage, 100% passing)
 - [x] Unit Testing in Spark Angular with `gulp test-angular` (100% coverage, 100% passing)
 - [x] Unit Testing in Spark React with `gulp test-react` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### CODE REVIEW

File: angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.ts [hunk 1]
```diff
@@ -18,11 +18,11 @@ import {
       <div [ngClass]="{ 'sprk-o-Box': dropdownType === 'mastheadSelector' }">
         <sprk-link
           [additionalClasses]="getTriggerClasses()"
-          ariaHasPopUp="true"
           (click)="toggle($event)"
           [idString]="idString"
           [analyticsString]="analyticsString"
-          role="combobox"
+          aria-haspopup="listbox"
+          [attr.aria-label]="title ? triggerText : screenReaderText"
```
Comment (RCopeland) [inline angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.ts]: if triggerText or screenReaderText aren't supplied, we should have a reasonable default

File: angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.ts [hunk 2]
```diff
@@ -64,12 +64,18 @@ import {
           </sprk-link>
         </div>
 
-        <ul class="sprk-c-Dropdown__links">
+        <ul
+          class="sprk-c-Dropdown__links"
+          role="listbox"
+          [attr.aria-label]="title ? title : screenReaderText"
```
Comment (RCopeland) [inline angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.ts]: same here, if title or screenReaderText aren't there, some kind of default should be used.

File: react/src/components/dropdown/SprkDropdown.js [hunk 3]
```diff
@@ -106,7 +106,8 @@ class SprkDropdown extends Component {
             additionalTriggerClasses,
           )}
           aria-expanded={isOpen}
-          role="listbox"
+          aria-haspopup="listbox"
+          aria-label={title ? triggerText : screenReaderText}
```
Comment (RCopeland) [inline react/src/components/dropdown/SprkDropdown.js]: same here, need a default if triggerText or screenReaderText arent supplied

File: react/src/components/dropdown/SprkDropdown.js [hunk 4]
```diff
@@ -146,7 +147,11 @@ class SprkDropdown extends Component {
                 <h2 className="sprk-c-Dropdown__title">{title}</h2>
               </div>
             )}
-            <ul className="sprk-c-Dropdown__links">
+            <ul
+              className="sprk-c-Dropdown__links"
+              aria-label={title ? title : screenReaderText}
```
Comment (RCopeland) [inline react/src/components/dropdown/SprkDropdown.js]: same here

File: react/src/components/dropdown/SprkDropdown.js [hunk 5]
```diff
@@ -188,6 +196,7 @@ class SprkDropdown extends Component {
                             'sprk-c-Dropdown__link--active': isActive,
                           })}
                           href={TagName === 'a' ? href || '#' : undefined}
+                          aria-label={content.title }
```
Comment (RCopeland) [inline react/src/components/dropdown/SprkDropdown.js]: extra space after the title

File: react/src/components/dropdown/SprkDropdown.test.js [hunk 6]
```diff
@@ -228,3 +228,41 @@ it('should not set href if the supplied tagname is not a', () => {
       .getAttribute('href'),
   ).toBe(null);
 });
+
+it('should choose correct aria-label text if default variant', () => {
```
Comment (RCopeland) [inline react/src/components/dropdown/SprkDropdown.test.js]: should test for when title or screenReaderText or triggerText arent supplied also

File: html/components/dropdown.stories.js [hunk 7]
```diff
@@ -75,13 +78,13 @@ export const informationalDropdown = () => {
   return `
     <a
       class="sprk-b-Link sprk-b-Link--plain sprk-u-mrs"
-      href="#nogo"
```
Comment (afebbraro) [inline html/components/dropdown.stories.js]: Why the removal of the href?

**Comment by afebbraro** [comment]
Hey @RVMendoza , in the description up above, could you explain what a11y changes needed to be made and why. Thank you!!

File: html/components/dropdown.stories.js (hunk 7 above)
Comment (RVMendoza) [reply html/components/dropdown.stories.js]: I think I got trigger happy -- addressed in this commit:
https://github.com/sparkdesignsystem/spark-design-system/pull/2391/commits/831877ca5930d31cc4c8902e98824d2637fe13c5

**Pull request merged**